ui/src/app/views/workspace-edit/component.ts

```diff
@@ -122,31 +123,65 @@ export class WorkspaceEditComponent implements OnInit {
     this.savingWorkspace = false;
   }
 
+  validateForm(): boolean {
+    if (this.savingWorkspace) {
+      return false;
+    }
+    this.nameNotEntered = isBlank(this.workspace.name);
+    if (this.nameNotEntered) {
+      return false;
+    }
+    return true;
+  }
+
+  addWorkspace(): void {
+    if (!this.validateForm()) {
+      return;
+    }
+    this.savingWorkspace = true;
+    this.errorHandlingService.retryApi(
+      this.workspacesService.createWorkspace(this.workspace))
+      .subscribe(
+        () => this.navigateBack(),
+        (error) => this.workspaceCreationError = true);
```

Comment:
In general, for code consistency, I would personally like us to surround all callbacks with {}, unless it is just calling a single other function (and maybe even then). I don't think we've discussed that yet, but imo it makes it easier to read?

Reply:
Done. Not sure I entirely agree on this point, but I think I can perhaps buy it for these side-effect callbacks. That said, I'd like to have a style guide that still allows the following:
`list.map(c => c+1);`

For at least lambda-like usage. I'm not sure there's a good line I can draw to capture those cases, but that's at least one point I'd like to maintain. Would prefer these issues were addressed by a style guide, but I don't see one readily available.

Reply:
I am perfectly happy with lambda-like usage for this, the piece that I would like to avoid is needing to remember the braces when adding code to the body of a callback, just because I worry about some errors popping up because of that. (Mostly around callbacks to API calls, where I can often see more code being added.)

I agree that we should have a style guide at some point. I know we had a style discussion a while back, but with the team changing it means that things are getting rehashed. (And I don't know if we even covered braces/callback convention)